public/js/user.js

```diff
@@ -277,10 +277,12 @@ function saveMarker(e) {
       url: "/marker" + (isUpdateRequest ? "/"+id : ""),
       type: isUpdateRequest?"PATCH":"POST",
       data: json,
-      dataType: "text",
+      dataType: "json",
+      processData: false,
+      contentType: "application/json",
     })
         .done(function(response) {
-          response = response.substring(1,response.length-1);
+          response = response.id;
```

Comment:
What's going on here? We've successfully created/updated the marker, the server has responded with it's representation of the marker, and we're throwing that away and only keeping the `id`? Why don't we update our (local) representation of the marker with what was returned?

I'm guessing the "old" API used to just return the ID, the API now returns the full marker resource, and instead of refactoring the code to use the resource returned, we're just hacking the response to look like the old response. If that's true, then we don't need to fix that here & now... but we should probably fix this (i.e., create a ticket to "Use the resource returned when creating or saving a marker").

Reply:
The server is only returning the id for now, but we can change that using a SELECT statement. The only new data that the server generates for now is the id (and the changed date, that will be fixed in ATLAS-165), so we're only keeping that.

Reply:
The server is only returning the id for now, but we can change that using a SELECT statement. The only new data that the server generates for now is the id (and the changed date, that will be fixed in ATLAS-165), so we're only keeping that.

Reply:
> I only returned the ID and didn't return whole resources in most of the POST and PATCH routes I wrote. Is it ok if I create a new issue to address that?

Yes. Let's create a new ticket for that. All REST calls that create or update resource(s) should respond with the corresponding resource(s). I'll go ahead and merge this PR.